packages/wasm/crate/src/tx.rs

```diff
@@ -173,34 +161,48 @@ pub fn build_parallel(
     Ok(tx.encode_to_vec())
 }
 
-/// Get transaction view, transaction perspective
-/// Arguments:
-///     full_viewing_key: `byte representation inner FullViewingKey`
-///     tx: `pbt::Transaction`
-///     idb_constants: IndexedDbConstants
-/// Returns: `TxInfoResponse`
-#[wasm_bindgen]
-pub async fn transaction_info(
+#[wasm_bindgen(typescript_custom_section)]
+const TRANSACTION_VIEW_PERSPECTIVE_TS: &'static str = r#"
+export function transaction_perspective_and_view(full_viewing_key: Uint8Array, tx: Uint8Array, idb_constants: any): Promise<{ txp: Uint8Array, txv: Uint8Array }>;
+"#;
+
+#[wasm_bindgen(skip_typescript)]
+pub async fn transaction_perspective_and_view(
     full_viewing_key: &[u8],
     tx: &[u8],
     idb_constants: JsValue,
-) -> WasmResult<JsValue> {
+) -> WasmResult<js_sys::Object> {
```

Comment:
wasm-bindgen support for async is rather bad, so i had to resort to a manual declaration. this is better than the alternatives